uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/M2ServletContextListener.java

```diff
@@ -149,16 +153,17 @@ private Properties readZipFile(String zipFilePath) {
         return fixNotMavenizedArtifact.getProperties(zipFilePath);
     }
 
-    public void deploy(final GAV gav,
-                       final String jarPath) {
+    private void deploy(final GAV gav,
+                        final String jarPath) {
 
         final Artifact artifact = MavenRepository.getMavenRepository().resolveArtifact(new AFReleaseIdImpl(gav.getGroupId(), gav.getArtifactId(), gav.getVersion()));
 
         if (artifact != null) {
+            logger.info("Maven Artifact {} already exists.", gav.toString());
```

Comment:
Please make this log message a bit more informative. Where does it exist (so user knows where he can delete it if needed)?